index.html

```diff
@@ -204,55 +205,2367 @@ <h3>
+              task=] on the [=relevant global object=] of [=this=] using the
+              [=HID device task source=] to [=reject=] |promise| with a
+              "{{NotAllowedError}}" {{DOMException}} and abort these steps.
+              </li>
+              <li>Invoke the operating system to send a feature report with the
+              specified |reportId:octet| and |data:BufferSource|.
+              </li>
+              <li>If this fails for any reason, [=queue a global task=] on the
+              [=relevant global object=] of [=this=] using the [=HID device
+              task source=] to [=reject=] |promise| with a "{{NetworkError}}"
+              {{DOMException}} and abort these steps.
+              </li>
+              <li>[=Queue a global task=] on the [=relevant global object=] of
+              [=this=] using the [=HID device task source=] to [=resolve=]
+              |promise| with `undefined`.
+              </li>
+            </ol>
+          </li>
+          <li>Return |promise|.
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h3>
+          <dfn>receiveFeatureReport()</dfn> method
+        </h3>
+        <div class="note">
+          <p>
+            The {{HIDDevice/receiveFeatureReport()}} method is used to request
+            a feature report with the specified |reportId:octet|. If the [=HID
+            interface=] does not [=use report IDs=], pass 0 instead of the
+            report ID.
+          </p>
+          <p>
+            For [=HID interfaces=] that [=use report IDs=], the report ID value
+            0 is reserved and should not be used.
+          </p>
+        </div>
+        <p>
+          The {{HIDDevice/receiveFeatureReport()}} method steps are:
+        </p>
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].
+          </li>
+          <li>If [=this=].{{HIDDevice/[[state]]}} is not `"opened"`, then
+          [=reject=] |promise| with a "{{InvalidStateError}}" {{DOMException}}
+          and return |promise|.
+          </li>
+          <li>If |reportId:octet| is 0 and the [=HID interface=] represented by
+          [=this=] [=uses report IDs=], or if |reportId| is not 0 and the
+          interface does not [=use report IDs=], then [=reject=] |promise| with
+          a "{{TypeError}}" {{DOMException}} and return |promise|.
+          </li>
+          <li>Append |promise| to
+          [=this=].{{HIDDevice/[[pendingReceiveFeatureReportPromises]]}}.
+          </li>
+          <li>Perform the following steps [=in parallel=]:
+            <ol>
+              <li>If |promise| is not in
+              [=this=].{{HIDDevice/[[pendingReceiveFeatureReportPromises]]}},
+              then abort these steps.
+              </li>
+              <li>Remove |promise| from
+              [=this=].{{HIDDevice/[[pendingReceiveFeatureReportPromises]]}}.
+              </li>
+              <li>Let |data:DataView| be the result of invoking the operating
```

Comment:
Is this where we want to leave a note that it is undefined whether or not _data_ will include a report ID? I forget what our eventual resolution on this question was.

Either way this should be explicit that _data_ is a DataView constructed over the bytes of the response.

Reply:
I think we decided to include the report ID byte because the data technically is from the device and for some devices may contain data other than the report ID.

I've added a note with the same comment from the Chromium implementation:

https://source.chromium.org/chromium/chromium/src/+/main:services/device/hid/hid_connection.h;l=69;drc=b3040fd7f4402e4658483321402dd342808361c8